### ISSUE TITLE
Error when checking target: expected activation_1 to have 3 dimensions

### ISSUE THREAD

**Issue by msaremi** [keras-team/keras#12602]
When I use the model with `categorical_crossentropy` and target values of shape `(batch_size, seq_size, vocab_size)` when running `fit_generator` the method runs fine. But when switching to `sparse_categorical_crossentropy` with target values of shape `(batch_size, seq_size, vocab_size)` the method raises this error:
    
    Error when checking target: expected activation_1 to have 3 dimensions, but got array with shape (20, 10)

Here is the model that I use:

    inp = Input(batch_shape=(batch_size, seq_size))
    mat = Embedding(num_classes, num_lstms)(inp)
    mat = LSTM(num_lstms, return_sequences=True, stateful=True)(mat)
    mat = LSTM(num_lstms, return_sequences=True, stateful=True)(mat)
    mat = Dropout(0.5)(mat)
    mat = TimeDistributed(Dense(vocab_size))(mat)
    out_1 = Activation('softmax')(mat)
    model = Model(inputs=[inp], outputs=[out_1])

**Comment by hr21** [comment]
Your input layer is recieving two parameters but it requires three check what are you passing through the layer (shapes of labels and inputs) are they of the desired format. if not consider adding a new channel to your inputs using np.newaxis(.,.,.,....) and add suitable parameter to the input in newaxis by performing numpy operations. That would do the job
